library/psa_crypto_slot_management.c

```diff
@@ -276,6 +277,76 @@ static psa_status_t psa_load_persistent_key_into_slot( psa_key_slot_t *slot )
 }
 #endif /* MBEDTLS_PSA_CRYPTO_STORAGE_C */
 
+#if defined(MBEDTLS_PSA_CRYPTO_BUILTIN_KEYS)
+
+static psa_status_t psa_load_builtin_key_into_slot( psa_key_slot_t *slot )
+{
+    psa_status_t status = PSA_ERROR_CORRUPTION_DETECTED;
+    psa_key_attributes_t attributes = PSA_KEY_ATTRIBUTES_INIT;
+    psa_key_lifetime_t lifetime = PSA_KEY_LIFETIME_VOLATILE;
+    psa_drv_slot_number_t slot_number = 0;
+    size_t key_buffer_size = 0;
+    size_t key_buffer_length = 0;
+
+    if( ! psa_key_id_is_builtin(
```

Comment:
I believe this check logically fits into the sub-function, and the sub-function is static/local to this compilation unit, so it doesn't matter from an optimisation standpoint. A static function with only one reference will almost always be inlined automatically as soon as you enable any level of optimisation.

Reply:
ok fine by me but I am not a fan of "the compiler will deal with it" type of argument. Why make the compiler's work harder and not organize the C code as we would like it to be executed ?

Reply:
I wasn't solely arguing the compiler will deal with it, I was only arguing that 'saving a function call' isn't a heavyweight argument. My main point was that I think the check belongs in the subroutine, just like in the call tree beneath `psa_load_persistent_key_into_slot` the ID is checked to be a persistent ID (in effect by checking whether it exists on disk).

In addition, the reason we're doing an explicit check for the built-in range is to avoid the platform from being able to override keys outside of the built-in range. So it's a piece of logic that IMO definitely belongs in the subroutine which handles interaction with the platform hook.

Reply:
Thanks for those additional explanations. I understand better now what you meant by "this check logically fits into the sub-function" which I was not sure about the meaning in your first reply.